twisted/test/proto_helpers.py

```diff
@@ -142,6 +142,13 @@ class StringTransport:
     @ivar io: A L{io.BytesIO} which holds the data which has been written to
         this transport since the last call to L{clear}.  Use L{value} instead
         of accessing this directly.
+
+    @ivar lenient: A L{bool} that controls whether L{StringTransport} will
```

Comment:
My biggest problem is with the lack documentation for this flag and the StringTransport.

As a normal dev, which has not read the ticket discussions, I will read this code and will wonder why do we not want to be `extraStrict` :)

So we can use the flag and a single class, but would help to also update the class docstring to include all the implementation details which were revealed in the Trac ticket and why the default implementation is more or less strict than the real world transports.

We already have StringTransportWithDisconnection which is using inheritance ... and by not using a flag it has the same signature as the real deal :)

I don't want to delay merging this code so I am OK with any fix.

I just wanted to leave my comments about how I would think that a 3rd party developer (not involved in this ticket) might perceive these changes